config/routes.js

```diff
@@ -1,32 +1,59 @@
-const axios = require('axios');
+const axios = require("axios");
+const bcrypt = require("bcryptjs");
+const { authenticate, generateToken } = require("../auth/authenticate");
 
-const { authenticate } = require('../auth/authenticate');
+const db = require("../database/dbConfig");
 
 module.exports = server => {
-  server.post('/api/register', register);
-  server.post('/api/login', login);
-  server.get('/api/jokes', authenticate, getJokes);
+  server.post("/api/register", register);
+  server.post("/api/login", login);
+  server.get("/api/jokes", authenticate, getJokes);
 };
 
 function register(req, res) {
   // implement user registration
+  const user = req.body;
+  user.password = bcrypt.hashSync(user.password, 12);
+
+  db.insert(user).then(ids => {
+    db.findById(ids[0])
+      .then(user => {
+        const token = generateToken(user);
+        res.status(201).json({ id: user.id, token });
+      })
+      .catch(err => {
+        res.status(500).send(err);
+      });
+  });
 }
 
 function login(req, res) {
   // implement user login
+  const creds = req.body;
+  db.getByUser(creds.username)
+    .then(user => {
+      // Checking to see username is valid, The Hash from clien === hash from db
```

Comment:
👍 love seeing comments like this. Your future self will thank you for these when you haven't looked at this code for a while.